### ISSUE TITLE
avoid different parameter names for plt.plot and plt.scatter  

### ISSUE THREAD

**Issue by gmamon** [matplotlib/matplotlib#20361]
In plt.plot, marker edge color is "markeredgecolor" or "mec". But in plt.scatter, marker edge color is "edgecolors". 
This difference in names is confusing! Could "mec" be recognized as edgecolors in plt.scatter?

**Comment by timhoffm** [comment]
Background: The reason for the difference is that `plot()` creates a `Line2D`, which can have markers on the data points and/or lines connecting these. Therefore `edgecolor` would be amgiguous for `plot()`. In contrast, `scatter()` has only markers and thus `edgecolors` is sufficient. It may or may not have been an intentional design.  Additionally, `scatter()` can color each marker individually. That's why it uses the plural `edgecolors`. Even though they have some overlap, `plot()` and `scatter()` have distinct sets of functionality, which shines through in their API.

The specific request to accept `mec` is somewhat hairy as well. Abbreviations such as `mec` are defined in the respective Artist class, here `LineCollection`. But `LineCollection` inherits from the general `Collection`, which only has `edgecolors` and `ec`. We would have to add `markeredgecolors` to `LineCollection` and define the abbreviation there. IMHO `mec` without `markedgecolor(s)` is not a viable option for consistency reasons. On the down side this means 1) we have even more ways to do the same thing 2) we need extra code to merge these ways to a single consistent meaning.

Note also that it doesn't stop at `mec`. This applies to all `marker*` parameters of `plot()`. Even more prominent, `linewidth` in `plot()` manipulates the connection lines, whereas `linewidths` in `LineCollection` manipulates what you would call `markeredgewidth`.

In summary, it is unfortunately not possible to make the APIs of `plot()` and `scatter()` more consistent. That ship has sailed long ago during the design of `LineCollection`. Any attempt to make them appear more similar will only result in other differences to be more prominent.

**Comment by dmatos2012** [comment]
Closing as reason for discrepancy was explained.

**Comment by NotNormallyAGitUser** [comment]
Is it possible to reopen this?  I don't see a reopen button.  I was going to suggest some harmonization measures by allowing (say) the scatter plot function to accept a keyword argument "markerfacecolor" as an *alias* for "facecolor" so that users can use the same terminology across plot types.  As an alias, it won't break backward compatibility.

**Comment by jklymak** [comment]
I don't think there is much appetite to reopen this.  Plot and scatter are very different methods that happen to have some visual overlap in a small subset of their functionality.  But their general functionality is much broader and therefore they have different kwargs suited to the general functionality rather than trying to make the small overlap consistent.

**Comment by timhoffm** [comment]
What could be investigated is how suitable the `PathCollection` interface is for `scatter()`. While it seems that `PathCollection` is only used for `scatter()` output, by name I would have expected a `MarkerCollection`. The first step would be to determine the interface for a potential `MarkerCollection`. The second step, check if/how we can integrate it in a backward compatible way (likely derive from `PathCollection`).

I make no claims whether this  is worth looking into or can work, but if any API changes were to be made, the should go into this direction.
